### PR TITLE
Upgrades and Updates

### PR DESCRIPTION
**Pull request by Arthur-Tesse**
Upgraded Kotlin to 1.1.51
Upgraded Android Platform to 27.0.0
Converted remaining Java classes to proper Kotlin ones
Fixed classes to be compliant with the upgrades
Updated my information in contributors.json

### CODE REVIEW

File: .idea/modules.xml [hunk 1]
```diff
@@ -3,7 +3,9 @@
   <component name="ProjectModuleManager">
     <modules>
       <module fileurl="file://$PROJECT_DIR$/NovelLibrary.iml" filepath="$PROJECT_DIR$/NovelLibrary.iml" />
+      <module fileurl="file://C:\Users\Arthur\Documents\Clones\NovelLibrary\NovelLibrary.iml" filepath="C:\Users\Arthur\Documents\Clones\NovelLibrary\NovelLibrary.iml" />
       <module fileurl="file://$PROJECT_DIR$/app/app.iml" filepath="$PROJECT_DIR$/app/app.iml" />
+      <module fileurl="file://C:\Users\Arthur\Documents\Clones\NovelLibrary\app\app.iml" filepath="C:\Users\Arthur\Documents\Clones\NovelLibrary\app\app.iml" />
     </modules>
```
Comment (gmathi) [inline .idea/modules.xml]: Please don't include this file.

File: app/src/main/java/io/github/gmathi/novellibrary/activity/NovelDetailsActivity.kt [hunk 2]
```diff
@@ -257,7 +257,7 @@ class NovelDetailsActivity : BaseActivity(), TextViewLinkHandler.OnClickListener
             .content(getString(R.string.confirm_remove_description))
             .positiveText(getString(R.string.remove))
             .negativeText(getString(R.string.cancel))
-            .icon(ContextCompat.getDrawable(this, R.drawable.ic_delete_white_vector))
+                .icon(ContextCompat.getDrawable(this, R.drawable.ic_delete_white_vector)!!)
```
Comment (gmathi) [inline app/src/main/java/io/github/gmathi/novellibrary/activity/NovelDetailsActivity.kt]: Is this showing a possible null pointer?

File: app/src/main/java/io/github/gmathi/novellibrary/activity/ReaderPagerActivity.kt [hunk 3]
```diff
@@ -215,8 +215,8 @@ class ReaderPagerActivity : BaseActivity(), ViewPager.OnPageChangeListener, Floa
 //    }
 
     override fun onBackPressed() {
-        if ((viewPager.adapter.instantiateItem(viewPager, viewPager.currentItem) as WebPageFragment).history.isNotEmpty())
-            (viewPager.adapter.instantiateItem(viewPager, viewPager.currentItem) as WebPageFragment).goBack()
+        if ((viewPager.adapter!!.instantiateItem(viewPager, viewPager.currentItem) as WebPageFragment).history.isNotEmpty())
+            (viewPager.adapter!!.instantiateItem(viewPager, viewPager.currentItem) as WebPageFragment).goBack()
         else
             super.onBackPressed()
     }
```
Comment (gmathi) [inline app/src/main/java/io/github/gmathi/novellibrary/activity/ReaderPagerActivity.kt]: I don't think we need those NPE & Optional (?) in this file

File: app/src/main/java/io/github/gmathi/novellibrary/network/NovelApi_Novel.kt [hunk 4]
```diff
@@ -6,15 +6,15 @@ import java.net.URI
...
+fun NovelApi.getNovelDetails(url: String): Novel {
     var novel: Novel? = null
     val host = URI(url).host
     when {
         host.contains(HostNames.NOVEL_UPDATES) -> novel = getNUNovelDetails(url)
         host.contains(HostNames.ROYAL_ROAD) -> novel = getRRNovelDetails(url)
         host.contains(HostNames.WLN_UPDATES) -> novel = getWlnNovelDetails(url)
     }
-    return novel
+    return novel!!
```
Comment (gmathi) [inline app/src/main/java/io/github/gmathi/novellibrary/network/NovelApi_Novel.kt]: We don't want a NPE When it fails to get novel details. Always handle it safely.

File: app/src/main/java/io/github/gmathi/novellibrary/model/Novel.kt [hunk 5]
```diff
@@ -30,25 +30,23 @@ class Novel : Serializable {
...
-            shortDescription = if (otherNovel.shortDescription != null) otherNovel.shortDescription else shortDescription
-            currentWebPageId = if (otherNovel.currentWebPageId != -1L) otherNovel.currentWebPageId else currentWebPageId
-            chapterCount = if (otherNovel.chapterCount != 0L) otherNovel.chapterCount else chapterCount
-            newChapterCount = if (otherNovel.newChapterCount != 0L) otherNovel.newChapterCount else newChapterCount
-            orderId = if (otherNovel.orderId != -1L) otherNovel.orderId else orderId
-
-            otherNovel.metaData.keys.forEach {
-                metaData.put(it, otherNovel.metaData[it])
-            }
+    fun copyFrom(otherNovel: Novel) {
```
Comment (gmathi) [inline app/src/main/java/io/github/gmathi/novellibrary/model/Novel.kt]: Always be prepared for the null variable being scenario

**Comment by Arthur-Tesse** [comment]
Most of the safety symbols were suggested by Android studio. I just followed his instructions to be 1.1.51 kotlin compliant

**Comment by gmathi** [comment]
Hey, that's a lot of work you put in. Thanks so much for taking your time to do it. I already had the project updated but forgot to sync previously. Could you pull the new changed from the 0.7.3 branch and resolve the conflicts by keeping the new code

**Comment by Arthur-Tesse** [comment]
Yes! I'll close this one and will fetch the new version source when you'll sync it here

**Pull request closed without merging**